### PR TITLE
Add Mining debug info

### PR DESCRIPTION
**Pull request by gdmmx**
Signed-off-by: gdmmx <gdmmx@163.com>

### Proposed changes in this pull request
Show mining debug info in getnodestate RPC if config.MiningDebug enable, helpful for unlucky miners guys.

### Type (put an `x` where ever applicable)
- [ ] Bug fix: Link to the issue
- [x] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### Checklist
Please put an `x` against the checkboxes. Write a small comment explaining if its `N/A` (not applicable)

- [x] Read the [CONTRIBUTION guidelines](https://github.com/nknorg/nkn#contributing).
- [x] All the tests are passing after the introduction of new changes.
- [ ] Added tests respective to the part of code I have written.
- [ ] Added proper documentation where ever applicable (in code and README.md).
- [x] Code has been written according to [NKN-Golang-Style-Guide](https://github.com/nknorg/nkn/wiki/NKN-Golang-Style-Guide)

### Extra information
Any extra information related to this pull request.

### CODE REVIEW

File: node/localnode.go [hunk 1]
```diff
@@ -66,6 +67,10 @@ func (localNode *LocalNode) MarshalJSON() ([]byte, error) {
 	out["uptime"] = time.Since(localNode.startTime).Truncate(time.Second).Seconds()
 	out["version"] = config.Version
 	out["relayMessageCount"] = localNode.GetRelayMessageCount()
+	if config.Parameters.MiningDebug {
+		out["proposalSubmitted"] = localNode.proposalSubmitted
```
Comment (yilunzhang) [inline node/localnode.go]: we should use GetProposalSubmitted() instead to prevent race condition

File: node/localnode.go [hunk 2]
```diff
@@ -207,6 +212,14 @@ func (localNode *LocalNode) maybeAddRemoteNode(remoteNode *nnetnode.RemoteNode)
 	return nil
 }
 
+func (localNode *LocalNode) GetProposalSubmitted() uint32 {
+	return localNode.proposalSubmitted
```
Comment (yilunzhang) [inline node/localnode.go]: GetProposalSubmitted() and IncrementProposalSubmitted() can be called together, so a lock is needed.

File: node/localnode.go [hunk 3]
```diff
@@ -207,6 +212,14 @@ func (localNode *LocalNode) maybeAddRemoteNode(remoteNode *nnetnode.RemoteNode)
...
+}
+
+func (localNode *LocalNode) IncrementProposalSubmitted() {
+	localNode.proposalSubmitted++
```
Comment (yilunzhang) [inline node/localnode.go]: Same as above

File: node/localnode.go (hunk 2 above)
Comment (gdmmx) [reply node/localnode.go]: I consider solidly consistency is unnecessary at here.
localNode.proposalSubmitted just only has one writer in a single goroutine consensus.startProposing(), and never should be in concurrent writing. (concurrent proposal is nonlicet and will be rejected by neighbors)
On the other hand, proposalSubmitted just a reference for evaluate node's propose success rate. Deviation 1 with probability is acceptable. Whatever, getnodeinfo RPC should not influenced the performance of consensus routine for avoid potential DDOS attack.

Comment (gdmmx) [reply node/localnode.go]: Anyway, I push a commit for make sure atomic R/W localNode.proposalSubmitted without lock.

**Pull request merged**